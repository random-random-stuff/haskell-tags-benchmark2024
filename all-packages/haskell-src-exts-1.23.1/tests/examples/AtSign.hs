{-# LANGUAGE TypeFamilies #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE TypeApplications #-}
{- |
Accelerate interface to the native CUDA implementation
of the Fourier Transform provided by the CUFFT library.
-}
module Data.Array.Accelerate.CUFFT.Private where


transform hndl@(Handle fallback mode width _) =
   wrap mode (A.constant width) $
   A.foreignAcc
      (AF.CUDAForeignAcc "transformForeign" $ transformForeign hndl)
      (unwrap mode (A.constant width) fallback)


