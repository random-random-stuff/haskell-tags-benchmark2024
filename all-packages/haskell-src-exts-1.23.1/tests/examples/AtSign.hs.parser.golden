ParseOk
  ( Module
      SrcSpanInfo
        { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 1 1 18 1
        , srcInfoPoints =
            [ SrcSpan "tests/examples/AtSign.hs" 1 1 1 1
            , SrcSpan "tests/examples/AtSign.hs" 2 1 2 1
            , SrcSpan "tests/examples/AtSign.hs" 3 1 3 1
            , SrcSpan "tests/examples/AtSign.hs" 8 1 8 1
            , SrcSpan "tests/examples/AtSign.hs" 8 1 8 1
            , SrcSpan "tests/examples/AtSign.hs" 11 1 11 1
            , SrcSpan "tests/examples/AtSign.hs" 18 1 18 1
            , SrcSpan "tests/examples/AtSign.hs" 18 1 18 1
            ]
        }
      (Just
         (ModuleHead
            SrcSpanInfo
              { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 8 1 8 49
              , srcInfoPoints =
                  [ SrcSpan "tests/examples/AtSign.hs" 8 1 8 7
                  , SrcSpan "tests/examples/AtSign.hs" 8 44 8 49
                  ]
              }
            (ModuleName
               SrcSpanInfo
                 { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 8 8 8 43
                 , srcInfoPoints = []
                 }
               "Data.Array.Accelerate.CUFFT.Private")
            Nothing
            Nothing))
      [ LanguagePragma
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 1 1 1 30
            , srcInfoPoints =
                [ SrcSpan "tests/examples/AtSign.hs" 1 1 1 13
                , SrcSpan "tests/examples/AtSign.hs" 1 27 1 30
                ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 1 14 1 26
                , srcInfoPoints = []
                }
              "TypeFamilies"
          ]
      , LanguagePragma
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 2 1 2 31
            , srcInfoPoints =
                [ SrcSpan "tests/examples/AtSign.hs" 2 1 2 13
                , SrcSpan "tests/examples/AtSign.hs" 2 28 2 31
                ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 2 14 2 27
                , srcInfoPoints = []
                }
              "TypeOperators"
          ]
      , LanguagePragma
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 3 1 3 34
            , srcInfoPoints =
                [ SrcSpan "tests/examples/AtSign.hs" 3 1 3 13
                , SrcSpan "tests/examples/AtSign.hs" 3 31 3 34
                ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 3 14 3 30
                , srcInfoPoints = []
                }
              "TypeApplications"
          ]
      ]
      []
      [ FunBind
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 1 15 48
            , srcInfoPoints = []
            }
          [ Match
              SrcSpanInfo
                { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 1 15 48
                , srcInfoPoints = []
                }
              (Ident
                 SrcSpanInfo
                   { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 1 11 10
                   , srcInfoPoints = []
                   }
                 "transform")
              [ PAsPat
                  SrcSpanInfo
                    { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 11 11 46
                    , srcInfoPoints =
                        [ SrcSpan "tests/examples/AtSign.hs" 11 15 11 16 ]
                    }
                  (Ident
                     SrcSpanInfo
                       { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 11 11 15
                       , srcInfoPoints = []
                       }
                     "hndl")
                  (PParen
                     SrcSpanInfo
                       { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 16 11 46
                       , srcInfoPoints =
                           [ SrcSpan "tests/examples/AtSign.hs" 11 16 11 17
                           , SrcSpan "tests/examples/AtSign.hs" 11 45 11 46
                           ]
                       }
                     (PApp
                        SrcSpanInfo
                          { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 17 11 45
                          , srcInfoPoints = []
                          }
                        (UnQual
                           SrcSpanInfo
                             { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 17 11 23
                             , srcInfoPoints = []
                             }
                           (Ident
                              SrcSpanInfo
                                { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 17 11 23
                                , srcInfoPoints = []
                                }
                              "Handle"))
                        [ PVar
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 24 11 32
                              , srcInfoPoints = []
                              }
                            (Ident
                               SrcSpanInfo
                                 { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 24 11 32
                                 , srcInfoPoints = []
                                 }
                               "fallback")
                        , PVar
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 33 11 37
                              , srcInfoPoints = []
                              }
                            (Ident
                               SrcSpanInfo
                                 { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 33 11 37
                                 , srcInfoPoints = []
                                 }
                               "mode")
                        , PVar
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 38 11 43
                              , srcInfoPoints = []
                              }
                            (Ident
                               SrcSpanInfo
                                 { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 38 11 43
                                 , srcInfoPoints = []
                                 }
                               "width")
                        , PWildCard
                            SrcSpanInfo
                              { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 44 11 45
                              , srcInfoPoints = []
                              }
                        ]))
              ]
              (UnGuardedRhs
                 SrcSpanInfo
                   { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 11 47 15 48
                   , srcInfoPoints =
                       [ SrcSpan "tests/examples/AtSign.hs" 11 47 11 48 ]
                   }
                 (InfixApp
                    SrcSpanInfo
                      { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 4 15 48
                      , srcInfoPoints = []
                      }
                    (App
                       SrcSpanInfo
                         { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 4 12 32
                         , srcInfoPoints = []
                         }
                       (App
                          SrcSpanInfo
                            { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 4 12 13
                            , srcInfoPoints = []
                            }
                          (Var
                             SrcSpanInfo
                               { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 4 12 8
                               , srcInfoPoints = []
                               }
                             (UnQual
                                SrcSpanInfo
                                  { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 4 12 8
                                  , srcInfoPoints = []
                                  }
                                (Ident
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 4 12 8
                                     , srcInfoPoints = []
                                     }
                                   "wrap")))
                          (Var
                             SrcSpanInfo
                               { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 9 12 13
                               , srcInfoPoints = []
                               }
                             (UnQual
                                SrcSpanInfo
                                  { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 9 12 13
                                  , srcInfoPoints = []
                                  }
                                (Ident
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 9 12 13
                                     , srcInfoPoints = []
                                     }
                                   "mode"))))
                       (Paren
                          SrcSpanInfo
                            { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 14 12 32
                            , srcInfoPoints =
                                [ SrcSpan "tests/examples/AtSign.hs" 12 14 12 15
                                , SrcSpan "tests/examples/AtSign.hs" 12 31 12 32
                                ]
                            }
                          (App
                             SrcSpanInfo
                               { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 15 12 31
                               , srcInfoPoints = []
                               }
                             (Var
                                SrcSpanInfo
                                  { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 15 12 25
                                  , srcInfoPoints = []
                                  }
                                (Qual
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 15 12 25
                                     , srcInfoPoints = []
                                     }
                                   (ModuleName
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 12 15 12 25
                                        , srcInfoPoints = []
                                        }
                                      "A")
                                   (Ident
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 12 15 12 25
                                        , srcInfoPoints = []
                                        }
                                      "constant")))
                             (Var
                                SrcSpanInfo
                                  { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 26 12 31
                                  , srcInfoPoints = []
                                  }
                                (UnQual
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 26 12 31
                                     , srcInfoPoints = []
                                     }
                                   (Ident
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 12 26 12 31
                                        , srcInfoPoints = []
                                        }
                                      "width"))))))
                    (QVarOp
                       SrcSpanInfo
                         { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 33 12 34
                         , srcInfoPoints = []
                         }
                       (UnQual
                          SrcSpanInfo
                            { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 33 12 34
                            , srcInfoPoints = []
                            }
                          (Symbol
                             SrcSpanInfo
                               { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 12 33 12 34
                               , srcInfoPoints = []
                               }
                             "$")))
                    (App
                       SrcSpanInfo
                         { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 13 4 15 48
                         , srcInfoPoints = []
                         }
                       (App
                          SrcSpanInfo
                            { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 13 4 14 69
                            , srcInfoPoints = []
                            }
                          (Var
                             SrcSpanInfo
                               { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 13 4 13 16
                               , srcInfoPoints = []
                               }
                             (Qual
                                SrcSpanInfo
                                  { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 13 4 13 16
                                  , srcInfoPoints = []
                                  }
                                (ModuleName
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 13 4 13 16
                                     , srcInfoPoints = []
                                     }
                                   "A")
                                (Ident
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 13 4 13 16
                                     , srcInfoPoints = []
                                     }
                                   "foreignAcc")))
                          (Paren
                             SrcSpanInfo
                               { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 14 7 14 69
                               , srcInfoPoints =
                                   [ SrcSpan "tests/examples/AtSign.hs" 14 7 14 8
                                   , SrcSpan "tests/examples/AtSign.hs" 14 68 14 69
                                   ]
                               }
                             (InfixApp
                                SrcSpanInfo
                                  { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 14 8 14 68
                                  , srcInfoPoints = []
                                  }
                                (App
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 14 8 14 44
                                     , srcInfoPoints = []
                                     }
                                   (Con
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 14 8 14 25
                                        , srcInfoPoints = []
                                        }
                                      (Qual
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan "tests/examples/AtSign.hs" 14 8 14 25
                                           , srcInfoPoints = []
                                           }
                                         (ModuleName
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan "tests/examples/AtSign.hs" 14 8 14 25
                                              , srcInfoPoints = []
                                              }
                                            "AF")
                                         (Ident
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan "tests/examples/AtSign.hs" 14 8 14 25
                                              , srcInfoPoints = []
                                              }
                                            "CUDAForeignAcc")))
                                   (Lit
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 14 26 14 44
                                        , srcInfoPoints = []
                                        }
                                      (String
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan "tests/examples/AtSign.hs" 14 26 14 44
                                           , srcInfoPoints = []
                                           }
                                         "transformForeign"
                                         "transformForeign")))
                                (QVarOp
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 14 45 14 46
                                     , srcInfoPoints = []
                                     }
                                   (UnQual
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 14 45 14 46
                                        , srcInfoPoints = []
                                        }
                                      (Symbol
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan "tests/examples/AtSign.hs" 14 45 14 46
                                           , srcInfoPoints = []
                                           }
                                         "$")))
                                (App
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 14 47 14 68
                                     , srcInfoPoints = []
                                     }
                                   (Var
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 14 47 14 63
                                        , srcInfoPoints = []
                                        }
                                      (UnQual
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan "tests/examples/AtSign.hs" 14 47 14 63
                                           , srcInfoPoints = []
                                           }
                                         (Ident
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan "tests/examples/AtSign.hs" 14 47 14 63
                                              , srcInfoPoints = []
                                              }
                                            "transformForeign")))
                                   (Var
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 14 64 14 68
                                        , srcInfoPoints = []
                                        }
                                      (UnQual
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan "tests/examples/AtSign.hs" 14 64 14 68
                                           , srcInfoPoints = []
                                           }
                                         (Ident
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan "tests/examples/AtSign.hs" 14 64 14 68
                                              , srcInfoPoints = []
                                              }
                                            "hndl")))))))
                       (Paren
                          SrcSpanInfo
                            { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 15 7 15 48
                            , srcInfoPoints =
                                [ SrcSpan "tests/examples/AtSign.hs" 15 7 15 8
                                , SrcSpan "tests/examples/AtSign.hs" 15 47 15 48
                                ]
                            }
                          (App
                             SrcSpanInfo
                               { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 15 8 15 47
                               , srcInfoPoints = []
                               }
                             (App
                                SrcSpanInfo
                                  { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 15 8 15 38
                                  , srcInfoPoints = []
                                  }
                                (App
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 15 8 15 19
                                     , srcInfoPoints = []
                                     }
                                   (Var
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 15 8 15 14
                                        , srcInfoPoints = []
                                        }
                                      (UnQual
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan "tests/examples/AtSign.hs" 15 8 15 14
                                           , srcInfoPoints = []
                                           }
                                         (Ident
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan "tests/examples/AtSign.hs" 15 8 15 14
                                              , srcInfoPoints = []
                                              }
                                            "unwrap")))
                                   (Var
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 15 15 15 19
                                        , srcInfoPoints = []
                                        }
                                      (UnQual
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan "tests/examples/AtSign.hs" 15 15 15 19
                                           , srcInfoPoints = []
                                           }
                                         (Ident
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan "tests/examples/AtSign.hs" 15 15 15 19
                                              , srcInfoPoints = []
                                              }
                                            "mode"))))
                                (Paren
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 15 20 15 38
                                     , srcInfoPoints =
                                         [ SrcSpan "tests/examples/AtSign.hs" 15 20 15 21
                                         , SrcSpan "tests/examples/AtSign.hs" 15 37 15 38
                                         ]
                                     }
                                   (App
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 15 21 15 37
                                        , srcInfoPoints = []
                                        }
                                      (Var
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan "tests/examples/AtSign.hs" 15 21 15 31
                                           , srcInfoPoints = []
                                           }
                                         (Qual
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan "tests/examples/AtSign.hs" 15 21 15 31
                                              , srcInfoPoints = []
                                              }
                                            (ModuleName
                                               SrcSpanInfo
                                                 { srcInfoSpan =
                                                     SrcSpan "tests/examples/AtSign.hs" 15 21 15 31
                                                 , srcInfoPoints = []
                                                 }
                                               "A")
                                            (Ident
                                               SrcSpanInfo
                                                 { srcInfoSpan =
                                                     SrcSpan "tests/examples/AtSign.hs" 15 21 15 31
                                                 , srcInfoPoints = []
                                                 }
                                               "constant")))
                                      (Var
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan "tests/examples/AtSign.hs" 15 32 15 37
                                           , srcInfoPoints = []
                                           }
                                         (UnQual
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan "tests/examples/AtSign.hs" 15 32 15 37
                                              , srcInfoPoints = []
                                              }
                                            (Ident
                                               SrcSpanInfo
                                                 { srcInfoSpan =
                                                     SrcSpan "tests/examples/AtSign.hs" 15 32 15 37
                                                 , srcInfoPoints = []
                                                 }
                                               "width"))))))
                             (Var
                                SrcSpanInfo
                                  { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 15 39 15 47
                                  , srcInfoPoints = []
                                  }
                                (UnQual
                                   SrcSpanInfo
                                     { srcInfoSpan = SrcSpan "tests/examples/AtSign.hs" 15 39 15 47
                                     , srcInfoPoints = []
                                     }
                                   (Ident
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan "tests/examples/AtSign.hs" 15 39 15 47
                                        , srcInfoPoints = []
                                        }
                                      "fallback"))))))))
              Nothing
          ]
      ]
  , [ Comment
        True
        (SrcSpan "tests/examples/AtSign.hs" 4 1 7 3)
        " |\nAccelerate interface to the native CUDA implementation\nof the Fourier Transform provided by the CUFFT library.\n"
    ]
  )
